### DISCUSSION TITLE
I wanted to know DgsDataLoader maxBatchsize option works?

### DISCUSSION THREAD

**Discussion by 3jin-p** [Netflix/dgs-framework#1439] [Q&A]
Thank you for providing a great open source for graphql. I have a question while using it.

For example, If I want to send data for 10,000 keys, but the data repository responds too slowly for a huge dataset

this case we set the maxBatchSize option to 100 is properly?

I wanted to know Does the dataloader work by processing 100 loads in parallel 100 times, or does all processing occur sequentially.

**Comment by srinivasankavitha** [comment]
It will process the keys in batches of 100 if that is the maxBatchSize. The
batches are processed in parallel at each level of processing in the graph.
These docs have a good example of how the batches are processed:
https://www.graphql-java.com/documentation/batching/

Hope that helps.

**Reply by 3jin-p** [reply]
Thank you for the information!
